### PR TITLE
Add the example 03 - Salesforce bulk processing example

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2258

Cannot enable the automatic build due to the Salesforce connector dependency.
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: examples/salesforce/main.bal [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    sbulk:BulkJob queryJob = check baseClient->createJob("insert", "Contact", "CSV");
+
+    do {
+        sbulk:BatchInfo batch = check baseClient->addBatch(queryJob, csvStream);
+        string batchId = batch.id;
+
+        // Get batch info
+        sbulk:BatchInfo batchInfo = check baseClient->getBatchInfo(queryJob, batchId);
+        io:println("Batch Info");
+        io:println(batchInfo);
```
Comment (daneshk) [inline examples/salesforce/main.bal:48]:
```suggestion
        io:println(string `Batch Info: ${batchInfo.toString()}`);
```

Comment (daneshk) [reply examples/salesforce/main.bal:48]: Shall we use log instead of io:println?

File: examples/salesforce/main.bal [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        foreach sbulk:Result res in batchResult {
+            if (!res.success) {
+                log:printError("Failed result, res=" + (res.errors is string ? <string>res.errors : ""));
+            }
+        }
+
+        // Close the job
+        _ = check baseClient->closeJob(queryJob);
+    } on fail error err {
+        log:printError(err.message());
```
Comment (daneshk) [inline examples/salesforce/main.bal:61]:
```suggestion
        log:printError("Salesforce batch operation failed", `error = err);
```

File: examples/salesforce/main.bal [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        // Get batch info
+        sbulk:BatchInfo batchInfo = check baseClient->getBatchInfo(queryJob, batchId);
+        io:println("Batch Info");
+        io:println(batchInfo);
+
+        // Get batch result
+        sbulk:Result[] batchResult = <sbulk:Result[]>check baseClient->getBatchResult(queryJob, batchId);
+        foreach sbulk:Result res in batchResult {
+            if (!res.success) {
+                log:printError("Failed result, res=" + (res.errors is string ? <string>res.errors : ""));
```
Comment (daneshk) [inline examples/salesforce/main.bal:54]:
```suggestion
                log:printError("Operation failed", result = (res.errors is string ? <string>res.errors : ""));
```

Comment (BuddhiWathsala) [reply examples/salesforce/main.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/466/commits/ff0cbcf4cf4c747731b3871d140928df5cd79a4d

File: examples/salesforce/main.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply examples/salesforce/main.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/466/commits/ff0cbcf4cf4c747731b3871d140928df5cd79a4d

File: examples/salesforce/main.bal (hunk 1 above)
Comment (BuddhiWathsala) [reply examples/salesforce/main.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/466/commits/ff0cbcf4cf4c747731b3871d140928df5cd79a4d

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**